src/selector.rs

```diff
@@ -23,12 +24,15 @@ pub struct Selector {
 impl Selector {
     /// Parses a CSS selector group.
 
-    pub fn parse(
-        selectors: &'_ str,
-    ) -> Result<Self, cssparser::ParseError<'_, SelectorParseErrorKind<'_>>> {
+    pub fn parse(selectors: &'_ str) -> Result<Self, SelectorErrorKind> {
         let mut parser_input = cssparser::ParserInput::new(selectors);
         let mut parser = cssparser::Parser::new(&mut parser_input);
-        parser::SelectorList::parse(&Parser, &mut parser).map(|list| Selector { selectors: list.0 })
+        match parser::SelectorList::parse(&Parser, &mut parser)
```

Comment:
Maybe we should replace this `match` with this code: 

`parser::SelectorList::parse(&Parser, &mut parser)
            .map(|list| Selector { selectors: list.0 })
            .map_err(|e| SelectorErrorKind::from(e))`

Reply:
here `E0308` occurs as the `err`s' type will conflict

Reply:
Maybe you are referring to lines 44 and 45 in src/error.rs

Reply:
Nevermind, for some reason it didnt work when I just copy-pasted the code. I typed it manually and it works fine

Reply:
> Maybe you are referring to lines 44 and 45 in src/error.rs

yes 44 and 45 throw `E0308` on my machine

Reply:
Ik, see https://github.com/causal-agent/scraper/pull/95#discussion_r1052127735

Reply:
Better leave it as you wrote it in the first place

Reply:
> Maybe we should replace this `match` with this code:
> 
> `parser::SelectorList::parse(&Parser, &mut parser) .map(|list| Selector { selectors: list.0 }) .map_err(|e| SelectorErrorKind::from(e))`

Sorry I have totally misread this message, I'll be pushing the changes shortly

Reply:
I have pushed the changes ([`7f4cf0f`](https://github.com/causal-agent/scraper/pull/95/commits/7f4cf0fbe9b3be8ebb8a163273b6df247d2fbd45))